src/components/data-table/data-table.component.ts

```diff
@@ -248,40 +259,100 @@ export default class ZnDataTable extends ZincElement {
       </div>`;
   }
 
-  goToFirstPage() {
-    this.page = 1;
+  getQueryBuilder() {
+    if (!this.filters) return html``;
+
+    return html`
+      <zn-dropdown>
+        <zn-menu>
+          <zn-query-builder filters="${this.filters}" size="small" outline>
+          </zn-query-builder>
+          <zn-button @click=${this.queryData}
+                     color="primary"
+                     type="submit">
+            Search
+          </zn-button>
+        </zn-menu>
+        <zn-button slot="trigger" size="small" icon="filter_alt" icon-size="16" color="secondary">
+          Filter
+        </zn-button>
+      </zn-dropdown>
+    `;
+  }
+
+  queryData(event: ZnSubmitEvent) {
+    const submit = event.target as ZnButton;
+    const parent: HTMLElement = submit.parentElement!;
+    const queryBuilder: ZnQueryBuilder = parent.querySelector('zn-query-builder')!;
+    const queryString = queryBuilder.value as string;
+
+    if (queryString) {
+      this.filter = queryString;
+      this._dataTask.run().then(r => r);
+    }
+  }
+
+  getActions() {
+    const actions = [];
+
+    actions.push(html`
+      <zn-button @click=${this.clearSelectedRows}
+                 ?disabled="${this.selectedRows.length <= 0}"
+                 size="small"
+                 outline>
+        Clear Selection
+      </zn-button>`);
+
+    Object.values(ActionSlots).forEach((slot: string) => {
+      if (this.hasSlotController.test(slot)) {
```

Comment:
Need to make sure hasSlotController is aware of delete-action and create-action, currently only aware of modify-action